app/components/RequestList.tsx

```diff
@@ -123,36 +113,30 @@ const RequestList: React.SFC<RequestObj> = props => {
       </div>
 
       <ReactTable
-        data={props.requests}
+        data={props.tabRecord.requests}
         columns={columns}
         showPagination={true}
         showPaginationTop={false}
         showPaginationBottom={true}
         defaultPageSize={10}
-        page={props.PageDetails[props.tabId] ? props.PageDetails[props.tabId].currentPageNumber : 0}
-        pageSize={
-          props.PageDetails[props.tabId] ? props.PageDetails[props.tabId].currentRowSize : 10
-        }
+        page={props.tabRecord.PageDetails.currentPageNumber}
+        pageSize={props.tabRecord.PageDetails.currentRowSize}
         onPageChange={changedPageNo =>
-          props.handlePaginationChange(changedPageNo, props.tabId, "currentPageNumber")
+          props.handlePaginationChange(changedPageNo, props.currentTabId, "currentPageNumber")
         }
         onPageSizeChange={changedRowSize =>
-          props.handlePaginationChange(changedRowSize, props.tabId, "currentRowSize")
+          props.handlePaginationChange(changedRowSize, props.currentTabId, "currentRowSize")
         }
         collapseOnDataChange={false}
         SubComponent={row => (
           <InterceptForm
+            tabRecord={props.tabRecord}
```

Comment:
The structure of the store is as follows : 
```javascript
store : {
  currentTab: tabId,
  currentUrl: "",
    tabRecord : {
      tabId: {
        PageDetails: [],
        checkedReqs: {},
        contentType: {},
        enabledStatus: false,
        errorMessage: "",
        interceptStatus: "",
        isInterceptorOn: true,
        requests: [],
        responseData: {},
        responseError: {},
        responseText: {},
        statusCodes: {}
       }
    }
}
```

The `InterceptForm` component needs the `statusCodes`, `contentType`,  `responseData`, `responseError` and `responseText` props from the parent. 

The store has to be re-structured  isthe following way to achieve this  : 
```js
store : {
  currentTab: tabId,
  currentUrl: "",
    tabRecord : {
      tabId: {
        PageDetails: [],
        checkedReqs: {},
        enabledStatus: false,
        errorMessage: "",
        interceptStatus: "",
        isInterceptorOn: true,
        requests: [],
       },
    requestRecords : {
    [requestId] : {
    statusCodes : {},
    contentType : {},
    responseData : {},
    responseText : {},
    responseData : {}
    }
  }
}
```

Here, we can pass just the required props as follows:

```html
<InterceptForm requestRecord={props.requestRecords[row.requestId]} ... />
```
This is how it needs to be done right?

Reply:
Thats a good approach on requestRecords. Keep these kinda suggestions coming. 
I missed this actually.

Reply:
Also, if the store structure is refactored to include `requestRecords`, we could also convert `statusCodes`, `contentType`, `responseData` and `responseType` to string type instead of object.